pkg/config/engine/engine.go

```diff
@@ -0,0 +1,63 @@
+/**
+# Copyright 2024 NVIDIA CORPORATION
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+**/
+
+package engine
+
+import "strings"
+
+// GetBinaryPathsForRuntimes returns the list of binary paths for common rutimes.
+// The following list of runtimes is considered:
+//
+//	the default runtime, "runc", and "crun"
```

Comment:
This will result in a behavioural change for the `containerd` case where we prioritise `runc` and then the `default_runtime`. I thought I'd preserve that behaviour as `containerd` is widely used and we probably have several users depending on the current behaviour

Reply:
If we are okay with this behavioural change, we can go ahead with it. However, we might need to call it out in release notes

Reply:
Maybe you could provide more context as to why `crio` was being handled differently?

Reply:
I've made the change. I thought it's safer to do this is in `crio` since it is not used as much and even the drop-in config override has never really been used so far

Reply:
Thanks. Let's revisit if we see any unexpected behaviour in testing.